WordPress/src/main/java/org/wordpress/android/ui/posts/PostListAction.kt

```diff
@@ -91,5 +101,16 @@ fun handlePostListAction(
         is PostListAction.DismissPendingNotification -> {
             NativeNotificationsUtils.dismissNotification(action.pushId, activity)
         }
+        is PostListAction.CopyUrl -> {
+            try {
+                activity.clipboardManager!!.setPrimaryClip(
+                        ClipData.newPlainText("${action.post.id}", action.post.link)
+                )
+                action.showToast.invoke(action.messageSuccess)
+            } catch (e: Exception) {
```

Comment:
I wonder if there is a way to avoid the generic exception here (thought we use it in a few other places in the code when working with the clipboard) 🤔 
The official [documentation](https://developer.android.com/guide/topics/text/copy-paste) does not cover this either. Is there a specific flow that causes an exception we can document with a comment to justify the addition of the [baseline exception](https://github.com/wordpress-mobile/WordPress-Android/pull/15570/files#diff-fa877df4f6878e67c74a6d479124a0bc5342b194c75568b95269934118273a42R352)?

Reply:
I will research this more deeply!

To be honest at first glance I thought wrapping these calls in a try/catch is all about the NPE that could be thrown when clipboardManager is null, and I explored the option of catching a `NullPointerException` instead but of course we'll get the same complaint from Detekt - `TooGenericExceptionCaught`.

Yet there may be something more to it as I could grasp from a few searches, hope the internet will help more :)!

If a NPE is the only concern your suggestion will already improve the code, since we'll handle the null case properly!
If it's true that we may encounter other exceptions ideally we know what and document it in the code 👍 

Thanks a lot 🙇🏻‍♂️!!!

Reply:
While researching the possible exceptions I found:

- [This article from 2013](https://commonsware.com/blog/2013/08/08/developer-psa-please-fix-your-clipboard-handling.html) which points to [an issue in Android 4.3](https://issuetracker.google.com/issues/36976027) users were experiencing having apps crash when another app / service would listen to clipboard changes.
  Since we don't support anything below Android 7.0 I think this shouldn't be a concern.
- [This chromium issue from 2020](https://bugs.chromium.org/p/chromium/issues/detail?id=1123727) which mentions possible exceptions on certain devices.

Looking more closely into [the chromium PR to fix the reported bug](https://chromium.googlesource.com/chromium/src/+/542cb9cfcc927295615809b0c99917b09a219d9f/ui/android/java/src/org/chromium/ui/base/Clipboard.java#332) we can get more insights into the possible issue and their solution:
<details>
  <summary>chromium/src/ui/android/java/src/org/chromium/ui/base/Clipboard.java:332</summary>

```java
    private boolean setPrimaryClipNoException(ClipData clip) {
        final String manufacturer = Build.MANUFACTURER.toLowerCase(Locale.US);
        // See crbug.com/1123727, there are OEM devices having strict mode violations in their
        // Android framework code. Disabling strict mode for non-google devices.
        try (StrictModeContext ignored = manufacturer.equals("google")
                ? null
                : StrictModeContext.allowAllThreadPolicies()) {
            mClipboardManager.setPrimaryClip(clip);
            return true;
        } catch (Exception ex) {
            // Ignore any exceptions here as certain devices have bugs and will fail.
            showCopyToClipboardFailureMessage();
            return false;
        }
    }
```
</details>

By looking at the [blame for that code](https://chromium.googlesource.com/chromium/src/+blame/542cb9cfcc927295615809b0c99917b09a219d9f/ui/android/java/src/org/chromium/ui/base/Clipboard.java#332) we can also notice that the call was being wrapped into a try/catch block at least since 2013-12 and there was a comment inside the `catch` stating:
```
// Ignore any exceptions here as certain devices have bugs and will fail.
```

That's all I could find on the topic In a Monday evening's research 🙃.

I'd add the above comment in the `catch` block since it may be a possible scenario but I'm also open for feedback 🙇🏻‍♂️

Reply:
Thank you for your research on this @ovitrif. The resources you found and especially the chromium PR would be great for future reference 🙇 
[The comment you added](https://github.com/wordpress-mobile/WordPress-Android/pull/15570/commits/7398ab33b76a3177288e56f7d8a932530458ea55#diff-b88b70cdd11730f84fd0e57469231b42b2823575b31e6600c79783e19e0378f0R112) seems adequate to me 👍

Reply:
Thank you @antonis for looking into this!
I've added [a comment with the link to the chromium changeset](https://github.com/wordpress-mobile/WordPress-Android/pull/15570/commits/6ce9767b29a03b57e7cf602cae9af0bdb8b22c5d) for future reference 🙇🏻‍♂️